### ISSUE TITLE
canceling import file dialog

### ISSUE THREAD

**Issue by haritop** [jmoenig/Snap--Build-Your-Own-Blocks#1108]
If  you cancel the import file dialog, then when you click on the Snap Logo, it shows the file dialog again, instead of the snap logo menu.
The problem disappears when you click anywhere in the control bar, except the buttons.
Tested on Chrome(Windows and Android).
It does not happen on Firefox.

**Comment by jmoenig** [comment]
Yeah, it's a known issue, because Morphic isn't getting an event when the user clicks "Cancel". You can force-remove the (hidden) input element by resizing the browser window. (Resizing checks whether an orphaned input element still exists in which case it gets removed from the document). After resizing the browser window you can again click on the Snap-logo.

Did I mention I hate them browser DOM architecture? Having to work with "hidden" DOM elements all the time instead of being able to just use them as first-class objects without actually adding them as documents nodes is such a ridiculous, obnoxious, annoying and fundamentally wrong design. I don't even begin to understand why people like web programming. Must be a case of the Stockholm syndrome. Seriously, it's why I based Snap on Morphic instead of the DOM. It should suffice to *create* an object in order to talk to it. Instead, in so called web-programming, you have to make it visible and reachable and then sorta tweak its properties to "fake hide" it before you can even send it a message. Instead of Object Oriented this kind of kludging around could also be called Fake Oriented Programming. FOP. There, I just invented the next paradigm shift :-)

**Comment by liam4** [comment]
new paradigm shift +1 will change all web standars kthxbai

Just kidding. I totally agree with you!

**Comment by haritop** [comment]
FOP :-)

**Comment by SillyInventor** [comment]
Some kludge that could be added to "fix" this :(

Add at file selector creation:
document.body.onfocus = 
            function () {
                    document.body.removeChild(inp);
                    myself.filePicker = null;
                    document.body.onfocus = null;
                };

Concept from: http://stackoverflow.com/questions/4628544/how-to-detect-when-cancel-is-clicked-on-file-input
